File: ruby/jjim.rb
require 'pp'
require 'json'

require_relative 'classes'
require_relative 'makeGraph'


def lines(code)
  code
  .split("\n")
  .select{ |x| x.strip != '' } # No empty lines
  .map{ |line| Indentation.new(line) }
end

def graph(code)
  lines = lines(code)
  Grapher.make_graph(Line.new('', lines).get_children)
end

File: src/jjim.js
import fs from 'fs'

import figureIndentation from './figureIndentation'
import makeGraph from './makeGraph'


export const graph = code => {
  const lines = code
    .split('\n')
    .filter(x => x.trim() !== '') // No empty lines
    .map(line => {
      const withoutIndent = line.replace(/^ */, '')
      return {
        indentation: line.length - withoutIndent.length,
        line: withoutIndent,
      }
    })

  if (lines.length === 0) {
    return makeGraph([], 0)
  } else {
    return makeGraph(figureIndentation(lines, lines[0].indentation))
  }
}

//console.log(JSON.stringify(graphToReact(graph), undefined, 4))
